packages/hoppscotch-app/helpers/import-export/import/openapi.ts

```diff
@@ -177,6 +181,190 @@ const parseOpenAPIV3BodyFormData = (
+    return Array.of(primitivePlaceholder.value, primitivePlaceholder.value)
+
+  // If the type is not primitive, it is "array"
+  // items property is required if type is array
+  return Array.of(
+    generateExampleArrayFromOpenAPIV2ItemsObject(
+      items.items as OpenAPIV2.ItemsObject
+    )
+  )
+}
+
+const generateRequestBodyExampleFromOpenAPIV2BodySchema = (
+  schema: OpenAPIV2.SchemaObject
+): RequestBodyExampleType => {
+  if (schema.example) return schema.example as RequestBodyExampleType
+
+  const primitiveTypeExample = pipe(
+    schema,
+    O.fromPredicate(
+      flow(
+        getSchemaTypeFromSchemaObject,
+        O.map(isSchemaTypePrimitive),
+        O.getOrElseW(() => false) // No schema type found in the schema object, assume non-primitive
+      )
+    ),
+    O.map(getSampleEnumValueOrPlaceholder) // Use enum or placeholder to populate primitive field
+  )
+
+  if (O.isSome(primitiveTypeExample)) return primitiveTypeExample.value
+
+  const arrayTypeExample = pipe(
+    schema,
+    O.fromPredicate(
+      flow(
+        getSchemaTypeFromSchemaObject,
+        O.map(isSchemaTypeArray),
+        O.getOrElseW(() => false) // No schema type found in the schema object, assume type to be different from array
+      )
+    ),
+    O.map((schema) => schema.items as OpenAPIV2.ItemsObject),
+    O.map(generateExampleArrayFromOpenAPIV2ItemsObject)
+  )
+
+  if (O.isSome(arrayTypeExample)) return arrayTypeExample.value
+
+  return pipe(
+    schema,
+    O.fromPredicate(
+      flow(
+        getSchemaTypeFromSchemaObject,
+        O.map(isSchemaTypeObject),
+        O.getOrElseW(() => false)
+      )
+    ),
+    O.chain((schema) =>
+      pipe(
+        schema.properties,
+        O.fromNullable,
+        O.map(
+          (properties) =>
+            Object.entries(properties) as [string, OpenAPIV2.SchemaObject][]
+        )
+      )
+    ),
+    O.getOrElseW(() => [] as [string, OpenAPIV2.SchemaObject][]),
+    A.reduce(
+      {} as { [name: string]: RequestBodyExampleType },
+      (aggregatedExample, property) => {
+        const example = generateRequestBodyExampleFromOpenAPIV2BodySchema(
+          property[1]
+        )
+        aggregatedExample[property[0]] = example
+        return aggregatedExample
+      }
+    )
+  )
+}
+
+const getSchemaFromOpenAPIV2Parameter = (
+  parameter: OpenAPIV2.Parameter
+): OpenAPIV2.SchemaObject => parameter.schema
+
+const generateRequestBodyExampleFromOpenAPIV2Body = (
+  op: OpenAPIV2.OperationObject
+): string =>
+  pipe(
+    (op.parameters ?? []) as OpenAPIV2.Parameter[],
+    A.findFirst((param) => param.in === "body"),
+    O.map(
+      flow(
+        getSchemaFromOpenAPIV2Parameter,
+        generateRequestBodyExampleFromOpenAPIV2BodySchema
+      )
+    ),
+    O.getOrElse(() => "" as RequestBodyExampleType),
+    (requestBodyExample) => JSON.stringify(requestBodyExample, null, "\t") // Using a tab character mimics standard pretty-print appearance
```

Comment:
Something like this to generate a prettified JSON could be a really useful function for the entire codebase, you can move this function to `~/helpers/functional/json.ts` so other parts of the app can use this.